test/e2e/lib/components/guide-component.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * External dependencies
+ */
+import { By, Key } from 'selenium-webdriver';
+
+/**
+ * Internal dependencies
+ */
+import * as driverHelper from '../driver-helper.js';
+
+import AsyncBaseContainer from '../async-base-container.js';
+
+export default class GuideComponent extends AsyncBaseContainer {
+	constructor( driver ) {
+		super( driver, By.css( '.components-guide' ) );
+	}
+
+	async dismissModal( selector, waitOverride ) {
+		if (
+			await driverHelper.isEventuallyPresentAndDisplayed(
+				this.driver,
+				By.css( '.components-guide__container' ),
+				waitOverride
+			)
+		) {
+			try {
+				// Easiest way to dismiss it, but it might not work in IE.
+				await this.driver.findElement( By.css( selector ) ).sendKeys( Key.ESCAPE );
```

Comment:
If we know what component is being used (`<Guide />`) then is there a reason we are asking for a `selector` here because we should already know what the components selector is.

Reply:
The reason I opted for making the selector dynamic is for the edge case that two different modals appear. In that case we may want to dismiss one or the other by just specifying the selector like `.nav-unification-modal`. Since this is a very edge case, I am open to removing the selector and just use a `<Guide />` specific class.

Reply:
Yep I understand. It might be a good idea to make the selector optional then, and fallback to the components main selector like @getdave mentioned.

Reply:
Sure, made it optional here 9a92636

Reply:
If we're adding a dynamic selector, I think we should have it in consideration as well in other places:

https://github.com/Automattic/wp-calypso/blob/9a92636ccead8bef25eca8c4ce4e3eb9a97b28a5/test/e2e/lib/components/guide-component.js#L22

https://github.com/Automattic/wp-calypso/blob/9a92636ccead8bef25eca8c4ce4e3eb9a97b28a5/test/e2e/lib/components/guide-component.js#L33

https://github.com/Automattic/wp-calypso/blob/9a92636ccead8bef25eca8c4ce4e3eb9a97b28a5/test/e2e/lib/components/guide-component.js#L38